src/wizards/Wizard.tsx

```diff
@@ -0,0 +1,415 @@
+/**
+ * Copyright (c) 2018 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+import {Icon} from "../icon";
+import {classNames} from "../utils";
+import {ClassNames} from "./ClassNames";
+import {Button, ButtonState} from "../forms/button";
+import {VerticalNav} from "../layout/vertical-nav";
+import {NavLink, NavLinkType} from "../layout/nav";
+
+export type WizardStep = {
+    stepName: string;
+    stepComponent: React.ReactNode;
+    stepId: number;
+    stepCompleted: boolean;
+    customStepNav?: WizardStepNavDetails;
+    isStepValid?: Function /* This function should return boolen value. And use to determine if step is valid or not */;
+};
+
+export type WizardStepNavDetails = {
+    stepNavIcon?: string;
+    stepNavbadges?: React.ReactNode;
```

Comment:
Maybe name this more general in case we want to use a different component inline.  In case we decided to use something other than badges.

Reply:
Thanks for noticing that. Changed the variable name